os/arch/arm/src/lm3s6965-ek/src/up_boot.c

```diff
@@ -107,3 +112,86 @@ void tiva_boardinitialize(void)
 	board_led_initialize();
 #endif
 }
+
+
+#ifdef CONFIG_BOARD_INITIALIZE
+
+#if defined(CONFIG_QEMU_SRAM) || defined (CONFIG_QEMU_SDRAM)
+
+#define 	QEMU_SMARTFS_PARTITION_START	(0)
+#define 	QEMU_SMARTFS_PARTITION_SIZE	(1024 * 1024) /* 1MB */
```

Comment:
please check here, tiva using below as a whole flash size.
 #define TIVA_FLASH_NPAGES        256
 #define TIVA_FLASH_PAGESIZE      1024

Reply:
Now sflash use TIVA_FLASH_PAGESIZE * TIVA_FLASH_NPAGES = 256K
You set 1024 as a nblocks when call mtd_partition below.
This is something wrong.
To sum up,
1. Total Flash size 256*1024 = 256K
2. Erase Block size = 1024B
3. Number of Erase Block = 256
4. Page size = 1024B
But you defined number of page as a 1024
Please double check this.

Reply:
TIVA_FLASH_NPAGES is 128K not 256 when running under QEMU.

Please check the changes in " os/arch/arm/src/tiva/chip/lm3s_flash.h"

#if defined(CONFIG_QEMU_SRAM) || defined (CONFIG_QEMU_SDRAM)                        
/* These parts all support a 1KiB erase page size and a total FLASH memory size  
 * of 128MB or 128K pages.                                                          
 */                                                                              
#define TIVA_FLASH_NPAGES        (128*1024)

Reply:
128 also problem, total size is TIVA_FLASH_NPAGES * TIVA_FLASH_PAGESIZE.
See ioctl, 
geo->blocksize & geo->erasesize is TIVA_FLASH_PAGESIZE,(Bytes)
geo->neraseblocks is TIVA_VIRTUAL_NPAGES
It means whole of flash is consist of 1024 * 128 = 128K

but you calling below
mtd_partition(mtd, partoffset, partsize / geo.blocksize, partno);
firstblock = (partoffset : 0)
nblocks = partsize/geo.blocksize : (1024 * 1024) / 1024

How allocate 1024K in 128K flash?

Reply:
Flash size is 128MB in case of QEMU.
Page size is 1024B, Hence TIVA_FLASH_NPAGES = 128 * 1024. =128K.

#define TIVA_VIRTUAL_NPAGES (TIVA_FLASH_NPAGES  - CONFIG_TIVA_FLASH_STARTPAGE).

Where,
CONFIG_TIVA_FLASH_STARTPAGE = 65536 (64K th page). 

Hence,
TIVA_VIRTUAL_NPAGES = 64K

So from 0 - 64MB of flash is not used . From 64MB offset, 1MB flash is used as a smartfs partition.

Reply:
you don't understand what I mean..
If below selected by config value, mount will be failed.
#define TIVA_FLASH_NPAGES        256
I am saying that you had better use MTDIOC_GEOMETRY instead of define value, because NPAGES changes significantly based on config.

Reply:
This code(mtd partition & mount)  is enabled only if defined(CONFIG_QEMU_SRAM) || defined (CONFIG_QEMU_SDRAM).

And #if defined(CONFIG_QEMU_SRAM) || defined (CONFIG_QEMU_SDRAM),
then  #define TIVA_FLASH_NPAGES        (128*1024)

So How it will fail?

Reply:
ah.. ok then I didn't check this file fully, I didn't check whole code is in those configs.
Then It is acceptable.